src/crimson/osd/osd_operation_sequencer.h

```diff
@@ -26,14 +26,16 @@ namespace crimson::osd {
 // can keep an op waiting in the case explained above.
 class OpSequencer {
 public:
-  template <typename HandleT,
+  template <typename OpT,
+            typename HandleT,
             typename FuncT,
             typename Result = std::invoke_result_t<FuncT>>
   seastar::futurize_t<Result>
-  start_op(HandleT& handle,
-           uint64_t prev_op,
-           uint64_t this_op,
+  start_op(const OpT& op,
```

Comment:
can we print op at the caller site?

Reply:
There are several options:

1. We can print `op` in `ClientRequest` and dump the `OpSequencer`-related details in `start_op()`. This would require 2 debugs.
2. We can dump everything at the caller side but the `OpSequencer` details would need to be exposed externally.
3. Just like now: `start_op()` takes the entire `op` and uses getters to extract the necessary data. The downside is that the extractions don't happen at the caller site anymore.
4. Introduce a public, `op`-taking variant of `start_op()` that would do the extractions and pass `prev_op` and `this_op` to a privatized `start_op` overload.

Reply:
i'd prefer going with option 1 and 2. do you think we can achieve the goal or get closer to it with the help of #41814?

Reply:
It's helpful when it comes to dumping the instance (`*this`) state but doesn't help with a method's local state (e.g. which branch has been taken). Of course, in the case  of`OpSequencer` the local states can be reconstructed as we would know the both values of inputs and the instance state as well. However, this could be a bit inconvenient in grepping over a bulky log (like the one from thrashing tests).

Reply:
i think in the context of debugging `OpSequencer`, what matters is `op.id` and other member variables of `OpSequencer`. probably that's why i don't understand why we need to print `op`. but neither do i want to stand in your way of doing so.